interactive mode to scripts + turning java installer to single step (#52)
conjurdemos/conjur-intro: demos/openshift-install/README.md

                kubernetes v1.11.0+d4cacc0  
                features: Basic-Auth  

## Important Note:
The commands below can run in either an interactive mode as well as in a command line mode.
Interactive mode means that each parameter that was not filled in the command line will be asked to be provided interactively.  
In interactive mode, some of these parameters have default values defined.
If the user does not provide an input, the default value is taken.  

> means that the user wants to run installer.sh without loading configuration.  

Reviewer: ```suggestion
```

Author: right - removing it

                kubernetes v1.11.0+d4cacc0  
                features: Basic-Auth  

## Important Note:

> Below commands can run in iteractive mode as well as in command line mode.  

Reviewer: ```suggestion
The commands below can run in either an interactive mode as well as in a command line mode.
```

Author: done

Interactive mode means that each parameter that was not filled in the command line will be asked to be provided interactively.  

> Some of these parameter have default values mentioned in interactive mode -  

Reviewer: ```suggestion
In interactive mode, some of these parameters have default values defined. 
```

Author: done

> and if the user does not provide an input there the default value is taken  

Reviewer: ```suggestion
If the user does not provide an input, the default value is taken.
```

Author: done

> The only parameter that cannot be sent in interactive mode is --with-config, because its absence in command line  

Reviewer: I am not sure I understand this. Why can't we have another prompt that asks the user whether the configuration load should be included? Could be something like this:

```
Please specify whether Conjur should be automatically configured [y]
```

Author: because if the user did not run it with --with-config it means that he wants to run it without config

Reviewer: I think most likely, 95% of the people running the script would want the config. Because if you want to just install Conjur, you could use the Helm chart pretty easily. So I think most likely, people would accidentally miss the parameter, because they run the script without parameters, than people that actually wanted to run it without.

What do you think?

Author: OK, I will remove this param

$ cd <home-dir>/conjur-intro/demos/java-api-client
$ ./build.sh
```

> 2. Installing and running java client opn Open Shift:

Reviewer: ```suggestion
2. Installing and running a demo Java client in OpenShift:
```

Author: the first one change seems to be irrelevant because the run it is related to does not run java client, but only Conjur + CLI. The second one is fixed

```bash
$ ./java-client-installer.sh --ocp-url <ocp-url:port> --docker-url <openshift-docker-registry-url> --project-name <project-name> --account-name <account-name> --authenticator <authenticator>
```

| Flag | Meaning |
| ---- | ------- |
|`--ocp-url` | Sets the OpenShift cluster in which the demo Java client would be installed (should have the same value as above) |
|`--docker-url` | Sets the Docker registry into which the container image of the demo Java client would be pushed |
|`--project-name` | Sets the OpenShift project in which the demo Java client would be installed (should have the same value as above) |
|`--account-name` | Sets the Conjur account that would be used by the demo Java client (should have the same value as above) |
|`--authenticator` | Sets the authenticator name that would be used by the demo Java client (should have the same value as above) |

3. Verify that all pods are up and running by:

```bash
$ oc get pods
```

> 4. Checking output of Java client container on pod #4:

Reviewer: ```suggestion
4. Checking output of the demo Java client container in pod #4:
```

Author: done

```bash
$ oc logs <pod-name> -c my-conjur-java-client  
```

> It should show that the secret was retrieved properly.

Reviewer: ```suggestion
    It should show that the secret was retrieved properly.
```

Author: done
